python/pyspark/version.py

```diff
@@ -16,4 +16,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-__version__ = "3.0.0.dev0"
+__version__ = "3.0.0"
```

Comment:
This should be `3.0.0-preview`.
cc @HyukjinKwon

Reply:
BTW, PR description knows what is correct. :)
> Update the PySpark version from 3.0.0.dev0 to 3.0.0-preview

Reply:
The problem is that, in PySpark it would normalize the version to `3.0.0rc0`, which would in turn lead to pip packaging failure, related logs can be found:
```
/tmp/tmp.RRgnNxjsCA/3.5/lib/python3.5/site-packages/setuptools/dist.py:398: UserWarning: Normalizing '3.0.0.preview' to '3.0.0rc0'
......
Requirement '/home/jenkins/workspace/SparkPullRequestBuilder@2/python/dist/pyspark-3.0.0.preview.tar.gz' looks like a filename, but the file does not exist
Processing ./python/dist/pyspark-3.0.0.preview.tar.gz
Could not install packages due to an EnvironmentError: [Errno 2] No such file or directory: '/home/jenkins/workspace/SparkPullRequestBuilder@2/python/dist/pyspark-3.0.0.preview.tar.gz'

```

Reply:
Oh. Got it.

Reply:
cc @holdenk since we need to publish this to https://pypi.org/project/pyspark/ and she is the expert on that.
I'm not sure this works or not there.

Reply:
Hm, I thought we're not going to publish it to PyPi. Are we going to do it on CRAN as well then?

Reply:
If we're going to do that, we should fix pip check or CRAN to allow other version formats. We cannot upload 3.0.0 and overwrite later ..

Reply:
Do you guys know if we're going to release on CRAN and PyPi, @jiangxb1987, @dongjoon-hyun, @holdenk, @srowen, @felixcheung, @rxin? If we're going to do that, let me try to take a stab to fix both.